tcx/src/handler.rs

```diff
@@ -335,34 +334,33 @@ pub fn private_key_store_export(data: &[u8]) -> Result<Vec<u8>> {
     let param: PrivateKeyStoreExportParam =
         PrivateKeyStoreExportParam::decode(data).expect("private_key_store_export");
     let mut map = KEYSTORE_MAP.write().unwrap();
+
     let keystore: &mut Keystore = match map.get_mut(&param.id) {
         Some(keystore) => Ok(keystore),
         _ => Err(format_err!("{}", "wallet_not_found")),
     }?;
 
-    if keystore.verify_password(&param.password) {
-        let pk_hex = keystore.export()?;
+    keystore.unlock_by_password(&param.password);
 
-        // private_key prefix is only about chain type and network
-        let coin_info = coin_info_from_param(&param.chain_type, &param.network, "")?;
-        let value = if param.chain_type.as_str() == "TRON" {
-            Ok(pk_hex.to_string())
-        } else {
-            let bytes = hex::decode(pk_hex.to_string())?;
-            let typed_pk = TypedPrivateKey::from_slice(CurveType::SECP256k1, &bytes)?;
-            typed_pk.fmt(&coin_info)
-        }?;
+    let pk_hex = keystore.export()?;
 
-        let export_result = KeystoreCommonExportResult {
-            id: keystore.id(),
-            r#type: KeyType::PrivateKey as i32,
-            value,
-        };
-
-        encode_message(export_result)
+    // private_key prefix is only about chain type and network
+    let coin_info = coin_info_from_param(&param.chain_type, &param.network, "")?;
+    let value = if param.chain_type.as_str() == "TRON" {
+        Ok(pk_hex.to_string())
     } else {
-        Err(format_err!("{}", "password_incorrect"))
-    }
+        let bytes = hex::decode(pk_hex.to_string())?;
+        let typed_pk = TypedPrivateKey::from_slice(CurveType::SECP256k1, &bytes)?;
+        typed_pk.fmt(&coin_info)
+    }?;
+
+    let export_result = KeystoreCommonExportResult {
+        id: keystore.id(),
+        r#type: KeyType::PrivateKey as i32,
```

Comment:
why not u32?

Reply:
It will never bigger than i32, if someone pass a -1, cast to u32 will get a very big and not readable value.